pennylane/ops/op_math/controlled.py

```diff
@@ -368,7 +368,7 @@ def map_wires(self, wire_map: dict):
         new_control_wires = Wires([wire_map.get(wire, wire) for wire in self.control_wires])
         new_work_wires = Wires([wire_map.get(wire, wire) for wire in self.work_wires])
 
-        return self.__class__(
+        return Controlled(
```

Comment:
why did you need to make this change?

Reply:
These calls to `self.__class__` create issues whenever we change the call signature of the child classes. For instance, in `CZ` we no longer need to provide a `base` argument, which results in an error. I changed this to `Controlled` because this way we have a consistent behaviour regardless of the call signatures.

Reply:
this could cause other operations to have problems, so I'd like to avoid this if possible. For example, if we later define `class ControlledWithCustomization(Controlled)`, then call `map_wires` on an instance of that, it will use the wrong constructor. I'm wondering why this was needed here but not in the `CY` PR - missing tests there, or maybe something here?

Reply:
Chatted offline about this. I opened #4139 to update these class methods to use `qml.ctrl` instead. Once that's in, we'll update this PR to just have custom logic for CZ in `ctrl`

Reply:
What's the issue with overriding these 4 methods in the subclass?

Reply:
it's fine to do, but we'll have to do it for CY as well. And maybe other controlled ops will want custom code in some other way, so the linked PR will make that easier. Plus, we don't want to re-write code that already works just because the interface is a little too unique/snowflakey for the new class

Reply:
I agree that it's the same issue for both `CY` and `CZ` (and other non-parametric ops), but that doesn't mean we have to override them more than once.

 I was thinking we could have a subclass of `Controlled`, say `ControlledNonParametric`, that would be the base class for `CY` and `CZ` and overrides these 4 methods. Then nothing needs to be changed in the `Controlled` class.